nulib/dc-nextjs
# 3054 google analytics

## What does this PR do?
- Adds GTM tracking code to the app
- Adds route specific data to GTM's `dataLayer`, which can get passed in via GTM up to Google Analytics

## To Test
(Assuming credentials are in place)
1. Run the app locally (or in AWS)
2. Open your Google Tag Manager account and select this container:

![image](https://user-images.githubusercontent.com/3020266/194571539-6cf6b921-5f49-42e2-96e0-2813e34bb178.png)

3. In Chrome browser, click the Preview button and this will fire up a local testing environment where you can see what tags and triggers are getting fired where.  Along with current state of the `dataLayer`.

![image](https://user-images.githubusercontent.com/3020266/194572127-507a55ab-d79a-4113-b41a-c0c0cf0abbf9.png)
